### PR TITLE
EVG-13974: Update status diff on the task page to support cedar test results

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-13974

I ran a patch resmoke patch build and realized that the current status diff on the patch task page does not support cedar test results. This fixes it.

@khelif96 I am not sure if this is also the case on the spruce UI, so just wanted to bring it to your attention in case it is.

### CODE REVIEW

File: model/task/task.go [hunk 1]
```diff
@@ -296,6 +296,14 @@ type TestResult struct {
 	LogTestName string `json:"log_test_name" bson:"log_test_name"`
 }
 
+func (tr *TestResult) TestName() string {
```
Comment (Kimchelly) [inline model/task/task.go:299]: Can you add a doc comment what this mean? It's not clear at all to a casual reader what the difference is between a "test name", a "log test name", and  "test file".

File: model/status_diff.go [hunk 2]
```diff
@@ -161,5 +169,35 @@ func StatusDiffTasks(original *task.Task, patch *task.Task) TaskStatusDiff {
...
+	originalMap := make(map[string]task.TestResult)
+	for _, test := range original {
+		originalMap[test.TestName()] = test
+	}
+
+	for _, test := range patch {
+		baseTest := originalMap[test.TestName()]
+
+		// Get the base name for windows/non-windows paths.
+		testName := path.Base(strings.Replace(test.TestName(), "\\", "/", -1))
```
Comment (Kimchelly) [inline model/status_diff.go:193]: Is this supposed to be `filepath.Base`? `path.Base` is for URLs.

Alternatively, you could just use `util.ConsistentFilepath`.

File: model/status_diff.go [hunk 3]
```diff
@@ -108,17 +109,24 @@ func StatusDiffBuilds(original, patch *build.Build) (BuildStatusDiff, error) {
...
 // getTestUrl returns the correct relative URL to a test log, given a
 // TestResult structure
 func getTestUrl(tr *task.TestResult) string {
-	// Return url if it exists. If there is no test, return empty string.
-	if tr.URL != "" || tr.LogId == "" { // If LogId is empty, URL must also be empty
+	if tr.URL != "" {
 		return tr.URL
 	}
-	return TestLogPath + tr.LogId
+	if tr.LogId != "" {
```
Comment (Kimchelly) [inline model/status_diff.go:122]: I don't really understand this change. The original comment says if LogId is empty, URL must be empty. Why does this need to be changed?

**Review by Kimchelly — CHANGES_REQUESTED** [review]

File: model/status_diff.go (hunk 3 above)
Comment (julianedwards) [reply model/status_diff.go:122]: With cedar test logs log id will not exist and we use `task_id/execution/test_name` in the `test_log` query.

https://github.com/evergreen-ci/evergreen/blob/main/service/ui.go#L325

File: model/status_diff.go (hunk 2 above)
Comment (julianedwards) [reply model/status_diff.go:193]: I copy/pasted this from StatusDiffTasks. It works and this is going to be gone once we completely switch to the spruce UI.

File: model/task/task.go (hunk 1 above)
Comment (julianedwards) [reply model/task/task.go]: I actually need two separate functions (one for the log test name and one for the display test name). Added both with better names and documentation.

**Review by Kimchelly — APPROVED** [review]

**Pull request merged**